sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -2125,6 +2266,65 @@ node nodeA = myGraph.add([&](handler& cgh) {
 dynParamAccessor.update(bufferB.get_access());
 ----
 
+=== Dynamic Event Update
+
+Example which shows how to create a dependency between two graphs using dynamic
+events which are then updated between executions.
+
+[source,c++]
+----
+...
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+queue myQueue;
+auto myContext = myQueue.get_context();
+auto myDevice = myQueue.get_device();
+
+// Create the first graph
+sycl_ext::command_graph graphA(myContext, myDevice);
+
+// Add a node to graphA
+sycl_ext::node nodeA = graphA.add(...);
+
+sycl_ext::command_graph<sycl_ext::graph_state::executable> execGraphA = 
+  graphA.finalize();
+
+// Get an event for the next execution of nodeA in execGraphA
+event eventA = execGraph.get_event(nodeA);
+
+// Create the second graph
+sycl_ext::command_graph graphB(myContext, myDevice);
+
+// Create the dynamic event which will wrap eventA and use it as a dependency
+// for a node in graphB
+sycl_ext::dynamic_event dynEvent{eventA};
+
+auto nodeB = graphB.add([&](handler& CGH){
+  CGH.depends_on(dynEvent);
+  CGH.parallel_for(...);
+});
+
+auto execGraphB = graphB.finalize({sycl_ext::property::graph::updatable});
+
+// Submit both graphs for execution
+myQueue.submit(execGraphA);
+myQueue.submit(execGraphB);
+myQueue.wait_and_throw();
+
+// Update the dynamic event with a new event frome execGraphA to reflect the
+// next execution
+dynEvent.update(execGraphA.get_event(nodeA));
+// Update execGraphB with the affected node to reflect those changes
+execGraphB.update(nodeB);
+
+// Both graphs can now be executed again with updated dependencies between them.
+myQueue.submit(execGraphA);
+myQueue.submit(execGraphB);
+myQueue.wait_and_throw();
+----
+
```

Comment:
Yes that would be the intended way of using it in record & replay, although that raises a good point that I need to add wording about extending `ext_oneapi_submit_barrier` to allow passing `dynamic_event`s instead of normal events.

Reply:
I have one question about `e1` used in `sycl::dynamic_event de{e1};`, it is generated within the graph recording stage (before g.finalize() is called).

while in the code below for the explicit API case, `eventA`, used in `sycl_ext::dynamic_event dynEvent{eventA};` is generated with `event eventA = execGraph.get_event(nodeA);` (after g.finalize() is called).

I'm wondering if `e1` and `eventA` have the same underlying implementation? (or they are two different concepts?)